lib/ui/recommendation-row.js

```diff
@@ -256,8 +256,10 @@ RecommendationRow.prototype = {
     const val = key.split('.').reduce(function(o, x) {
       return (typeof o == 'undefined' || o === null) ? o : o[x];
     }, this._data);
-    const isEmptyObject = (typeof val == 'object' && !Object.keys(val).length);
-    return (typeof val == 'undefined' || !val || isEmptyObject) ? null : val;
+    let isEmptyObject = val => {
+      return val && typeof val == 'object' && !Object.keys(val).length;
+    };
+    return (typeof val == 'undefined' || !val || isEmptyObject(val)) ? null : val;
```

Comment:
The `typeof val !== 'undefined'` check is only needed if the variable `val` might be uninitialized, but since it's defined as a const above, you don't need it.

I think you can get away with just doing this:
```js
return !!val && !isEmptyObject(val) && val || null;
```